static/app/components/nav/utils.tsx

```diff
@@ -0,0 +1,170 @@
+import type {ComponentProps} from 'react';
+import type {LocationDescriptor} from 'history';
+
+import type Feature from 'sentry/components/acl/feature';
+import {isItemActive} from 'sentry/components/sidebar/sidebarItem';
+import {SIDEBAR_NAVIGATION_SOURCE} from 'sentry/components/sidebar/utils';
+import normalizeUrl from 'sentry/utils/url/normalizeUrl';
+import type {useLocation} from 'sentry/utils/useLocation';
+
+/** NavItem is the base class for both SidebarItem and SubmenuItem  */
+interface NavItem {
+  /** User-facing item label, surfaced in the UI. Should be translated! */
+  label: string;
+  /** Optionally, props which should be passed to a wrapping `<Feature>` guard */
+  feature?: Omit<ComponentProps<typeof Feature>, 'children'>;
+}
+
+/** NavItems are displayed in either `main` and `footer` sections */
+export interface NavItemLayout<Item extends NavItem> {
+  main: Item[];
+  footer?: Item[];
+}
+
+/** SidebarItem is a top-level NavItem which is always displayed in the app sidebar */
+export interface NavSidebarItem extends NavItem {
+  /** The icon to render in the sidebar */
+  icon: JSX.Element;
```

Comment:
```suggestion
  icon: JSX.Element;
```
Question: does this need to be a JSX.Element or could it be a React.ReactNode?
If we dont want falsy icon values, it could be something like NonNullable<React.ReactNode>